src/recipe.rs

```diff
@@ -247,15 +247,24 @@ impl<'src, D> Recipe<'src, D> {
       message: format!("bad shebang line: {}", shebang_line),
     })?;
 
-    let tmp = tempfile::Builder::new()
-      .prefix("just")
-      .tempdir()
-      .map_err(|error| Error::TmpdirIo {
-        recipe: self.name(),
-        io_error: error,
-      })?;
-    let mut path = tmp.path().to_path_buf();
+    let tmp = if let Some(temp_dir) = &context.settings.temp_dir {
+      tempfile::Builder::new()
+        .tempdir_in(temp_dir.raw)
+        .map_err(|error| Error::TmpdirIo {
+          recipe: self.name(),
+          io_error: error,
+        })?
+    } else {
+      tempfile::Builder::new()
+        .prefix("just")
+        .tempdir()
+        .map_err(|error| Error::TmpdirIo {
+          recipe: self.name(),
+          io_error: error,
+        })?
+    };
```

Comment:
Maybe something like this to deduplicate a little bit, and preserve the temporary directory prefix:

```suggestion
    let mut tempdir_builder = tempfile::Builder::new();
    
    tempdir_builder.prefix("just");

    let tempdir = match &context.settings.tempdir {
      Some(tempdir) => tempdir_builder.tempdir_in(tempdir),
      None => tempdir_builder.tempdir(),
    }.map_err(|error| Error::TmpdirIo {
      recipe: self.name(),
      io_error: error,
    })?;
```